helpers/calendar/analytics_helper.py

```diff
@@ -177,16 +178,37 @@ def get_room_statistics(self, number_of_events_in_room, all_details):
         return result
 
     @staticmethod
-    def get_total_bookings(self, query, start_date, end_date):
+    def get_total_bookings(instance, query, start_date, end_date, room_id=None):
         bookings = 0
-        rooms = CommonAnalytics.get_room_details(self, query)
+        rooms = CommonAnalytics.get_room_details(instance, query)
+        active_rooms = RoomModel.query.filter(RoomModel.state == "active")
         for room in rooms:
-            all_events = CommonAnalytics.get_all_events_in_a_room(
-                self, room["room_id"], start_date, end_date)
-            if all_events:
-                bookings += len(all_events)
+            if room_id:
+                exact_room = active_rooms.filter(
+                    RoomModel.id == room_id).first()
+                if not exact_room:
+                    raise GraphQLError("Room Id does not exist")
+                if room.get('calendar_id') == exact_room.calendar_id:
+                    bookings = CommonAnalytics.get_bookings_count(instance,
+                                                                  room,
+                                                                  start_date,
+                                                                  end_date,
+                                                                  bookings)
+            else:
```

Comment:
@dnuwa I think you can get rid of this else statement and still have same functionality

Reply:
@ThaDeveloper when the `else` is removed each time this function is run, we'll have wrong data for the booking count of the single room because it will always return the bookings count for all the rooms. kindly try it out, you will see what I mean.

Reply:
@dnuwa I had tried it like below and got similar results
![image](https://user-images.githubusercontent.com/12385482/55639274-a9cbd500-57d1-11e9-875e-7f26397412d9.png)

Reply:
this was fixed. thanx @ThaDeveloper